src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -189,15 +189,45 @@ public class BlobFileStore : IFileStore
     }
 
     /// <inheritdoc />
-    public async Task DeleteFileIfExistsAsync(long version, string partitionName, CancellationToken cancellationToken)
+    public async Task DeleteFileIfExistsAsync(long version, Partition partition, FileProperties fileProperties, CancellationToken cancellationToken)
     {
+        EnsureArg.IsNotNull(partition);
 
-        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version, partitionName);
-        _logger.LogInformation("Trying to delete DICOM instance file with watermark '{Version}'.", version);
+        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version, partition, fileProperties);
```

Comment:
Just wanted to be sure, in some place we pass fileproperties as null explicitly, hope we dont get null reference